tools/measurement/latency/src/index.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as k8s from '@kubernetes/client-node';
+import * as fs from 'fs';
+import * as mkdirp from 'mkdirp';
+import * as path from 'path';
+import { argv } from 'yargs';
+
+import { TEST_TARGET_ID_MAP } from './test_target_id_map';
+import { ProwJob, ReplaceMeasurement, Report, Table } from './types';
+
+async function getProwJobs(api: k8s.Custom_objectsApi): Promise<ProwJob[]> {
+  const group = 'prow.k8s.io';
+  const version = 'v1';
+  const namespace = 'default';
+  const plural = 'prowjobs';
+
+  return new Promise((resolve, reject) => {
+    api
+      .listNamespacedCustomObject(group, version, namespace, plural)
+      .then(res => {
+        if (res.response.statusCode === 200) {
+          resolve(res.body.items as ProwJob[]);
+        } else {
+          const error = {
+            statusCode: res.response.statusCode,
+            statusMessage: res.response.statusMessage,
+          };
+          reject(`Query k8s cluster failed with error "${error}".`);
+        }
+      })
+      .catch(error => {
+        reject(`Query k8s cluter failed with error "${error}".`);
+      });
+  });
+}
+
+function isAndroidSdkJob(prowJob: ProwJob): boolean {
+  return TEST_TARGET_ID_MAP.has(prowJob.spec.job);
+}
+
+function constructReport(prowJobs: ProwJob[]): Report {
+  /* Temporarily allow snake case naming for report object assembly. */
+  /* tslint:disable:variable-name */
+  const table_name = 'TestResults';
+  const column_names = [
```

Comment:
What schema is this? This is different from the schema that I made.

Reply:
Right. I intend to make some change to the schema. The new column `test_id` (an UUID to specify a test run) might be most confusing. The reason is that we are querying prow regularly, and might get info for a prow job that we already got and uploaded from previous query. This column helps prevent us from uploading duplicate records.

What do you think? Cory and I discussed about it yesterday. You were not here so didn't get a chance to check with you.